doc/users/faq/howto_faq.rst

```diff
@@ -9,6 +9,80 @@ How-to
 .. contents::
    :backlinks: none
 
+
+_how-to-too-many-ticks
+
+Why do I have so many ticks, and/or why are they out of order?
+--------------------------------------------------------------
+
+Sometimes Matplotlib will unexpectedly plot with a tick for each data point,
+and/or the ticks will be out of numerical order. This is usually a sign that you
+have passed in a list of strings rather than a list or array of floats or
+datetime objects.  This will often happen when reading in a comma-delimited text
+file. Matplotlib treats lists of strings as "categorical" variables
+(:doc:`/gallery/lines_bars_and_markers/categorical_variables`), and by default
+puts one tick per "category", and plots them in the order in which they are
+supplied.
+
+In the example below, the upper row plots are plotted using strings for *x*;
+note that each string gets a tick, and they are in the order of the list passed
+to Matplotlib. In the lower row the data is converted to either floats or
+datetime64; note that the ticks are now ordered and spaced numerically.
+
+.. plot::
```

Comment:
I find the examples a bit extensive. Would the following be sufficient?

```
import matplotlib.pyplot as plt
import numpy as np

fig, ax = plt.subplots(1, 2, constrained_layout=True, figsize=(6, 2))

ax[0].set_title('Ticks seem out of order / misplaced')
x = ['5', '20', '1', '9']  # strings
y = [5, 20, 1, 9]
ax[0].plot(x, y, 'd')
ax[0].tick_params(axis='x', labelcolor='red', labelsize=14)

ax[1].set_title('Many ticks')
x = [str(xx) for xx in np.arange(100)]  # strings
y = np.arange(100)
ax[1].plot(x, y)
ax[1].tick_params(axis='x', labelcolor='red', labelsize=14)
```

![grafik](https://user-images.githubusercontent.com/2836374/148307524-84e1de33-1c99-4f8b-8f5c-d9b9e45a5bae.png)

Reply:
I see that you have updated the examples, and they are more clear now.

Still, do we need more than a screen full of figures and code to illustrate the relatively simple fact "passing string data plots all values as ticks in the given order"?

Motivation for the proposed plot above:
- I've limited this to the error cases. Plotting them is helpful to see how strings typically show up as "wrong" itics
- I've left out the respective "correct" plots with numbers for brevity. It should be quite clear from highlighted ticks that these are undesired. How the correct plot looks does not add much information IMHO.
- I've chosen two plots which show the characteristic error signatures "wrong order" and "too many ticks"
  "Dates out of order" is essentially the same as "numbers out of order".

I favor this to reduce clutter and keep the information density high. But I won't fight over it. If you feel strongly on having the correct plot as well, I suggest to at least remove the dates and use the same layout in both plots to make them equal in size.

Reply:
My apologies, I wrote a reply, but must have closed it before hitting save....

a) datetime-as-string is a _very_ common failure mode, so I think it should be in the FAQ
b) for beginners "turn your strings into floats" is going to require a google search;  half the time, even I have to look up how to do datetimes.  I felt it was important to offer a basic solution, and also to show the "expected" output so the "unexpected" is crystal clear. 

This section is for folks who are trying to do their first plot, reading their Excel data in as a csv.  While I agree we should be as concise as possible, I think here the extra guidance is worth the virtual real estate to get them on the right path.

Reply:
> This section is for folks who are trying to do their first plot, reading their Excel data in as a csv.

IMHO this can happen with all levels of experience.

Anyway, this is a judgement call on the audience and the amount of detail we think is necessary. I appreciate your position, but cannot sign-off on it. If others approve and merge I'm fine with that.

Reply:
I mean its a How-To - I would argue we should provide concrete solutions.

Reply:
It‘s not really a how-to, it‘s only in how-to because we don‘t have the proper category yet. https://github.com/matplotlib/matplotlib/pull/21943#issuecomment-994113006

Reply:
Let me rephrase: whatever we call it, I think it should have a solution in it, not just the problem.

Reply:
To me the solution is „don‘t pass in strings“. How to get from strings to numerical values depends on the data, and is not Matplotlibs business. I‘m fine that some people may have to do an additional google search. As said, it‘s a judgement call.

Reply:
OK, well, lets move the full example into an example.